arabic/unpacked/arabic.js

```diff
@@ -0,0 +1,448 @@
+      });
+    };
+
+    [
+      'mfrac',
+      'mi',
+      'mn',
+      'mo',
+      'mrow',
+      'ms',
+      'msqrt',
+      'msubsup',
+      'mroot',
+      'mtext'
+    ].forEach(makeElementFlippable);
+
+    MathJax.Hub.Register.StartupHook('HTML-CSS mtable Ready', function () {
+      makeElementFlippable('mtable');
+
+      MathJax.Hub.Startup.signal.Post('Arabic mtable Ready');
+    });
+
+
+    MathJax.Hub.Startup.signal.Post('Arabic Ready');
+  });
+});
+
+MathJax.Hub.Register.StartupHook('Arabic TeX Startup', function () {
+  var Text = MathJax.Extension.Arabic.Text;
+
+  MathJax.Hub.CombineConfig("Arabic", {
+      identifiersMap: {
+        // Math functions
+        'sin': 'جا',
+        'cos': 'جتا',
+        'tan': 'ظا',
+        'cot': 'ظتا',
+        'sec': 'قا',
+        'csc': 'قتا',
+        'log': 'لو'
+      },
+      // Limits
+      operatorsMap: {
+        'lim': 'نهــا'
+      }
+    });
+
+  MathJax.Extension.Arabic.config = MathJax.Hub.CombineConfig('Arabic', {
+    dict: {
+      // A macros to force English zero in both languages
+      "Zero": ["zero", Text('0', 'صفر')],  // Better localized Zero
+      "Radius": ["radius", Text('r', 'نق')],  // Circle radius
+      "Area": ["Area", Text('A', 'م')]  // Area of circles and other stuff
+    }
+  });
+});
+
+MathJax.Extension.Arabic = {
+  version: '1.0.0',
+  config: MathJax.Hub.CombineConfig("Arabic", {
+    dict: {},
+    isArabicPage: (document.documentElement.lang === 'ar'),
+    identifiersMap: {},
+    numbersMap: {
+      '0': '٠',
+      '1': '١',
+      '2': '٢',
+      '3': '٣',
+      '4': '٤',
+      '5': '٥',
+      '6': '٦',
+      '7': '٧',
+      '8': '٨',
+      '9': '٩'
+    },
+    operatorsMap: {
+      // English to Arabic punctuations
+      ',': '،',
+      ';': '؛'
+    }
+  }),
+  arabicLanguageRegExp: /([\u0600-\u06FF]+)/g,
+  TeX: function (english, arabic) {
+    // Creates a translated TeX macro.
+
+    return function (name) {
+      var TEX = MathJax.InputJax.TeX;
+
+      var tex;
+      if ('ar' === this.stack.env.lang) {
+        tex = arabic;
+      } else {
+        tex = english;
+      }
+
+      this.Push(TEX.Parse(tex).mml());
+    };
+  },
+  Text: function (english, arabicText) {
+    // Creates a translated TeX macro, with an Arabic plain text.
+
+    return MathJax.Extension.Arabic.TeX(english, '\\fliph{\\text{' + arabicText + '}}');
+  },
+  Symbols: function (english, arabicSymbols) {
+    // Creates a translated TeX macro that converts Arabic symbols into text nodes,
+    // and treats everything else as normal TeX.
+    var arabic = arabicSymbols.replace(
+      MathJax.Extension.Arabic.arabicLanguageRegExp,
+      '\\fliph{\\text{$1}}'
+    );
+
+    return MathJax.Extension.Arabic.TeX(english, arabic);
+  }
+};
+
+MathJax.Hub.Startup.signal.Post('Arabic TeX Startup');
+
+MathJax.Hub.Register.StartupHook('TeX Jax Ready', function () {
+  MathJax.Hub.Register.StartupHook('Arabic TeX Startup', function () {
```

Comment:
I've merged the two tex file in a single file, and this call is no longer necessary.

However, the signal is still there for other extensions.